### PR TITLE
New adaper structure in archetype

### PR DESCRIPTION
**Pull request by Nemikor**
# Description

Adding redesigned adapter structure to archetype

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Creating a new adapter and testing all SIP functionalities

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing (regression) unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by MartinBuchheim — DISMISSED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -39,16 +39,15 @@ systems, but all the principles apply equally to such scenarios.
 The image below shows the modules structure of a single SIP adapter.
 To provide a level of flexibility, SIP splits the problem into two, giving the possibility to develop and/or deploy connectors
 for individual systems separately. The integration
-logic is divided into three modules: **System A - Connector**, **System B - Connector** and **Domain**. The fourth module is the **Application**
-module. Its purpose is to converge the three integration modules into one deployable and executable application.
-The blue arrows represent dependencies of the application module.
+logic is divided into three packages: **System A - Connector**, **System B - Connector** and **Common - Domain**.
```
Comment (vladiIkor) [inline docs/README.md]: Changes should go to docs-snapshot folder, right?

File: changelogs/feature/new_adapter_structure.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "author": "Nemikor",
+  "pullrequestId": "106",
+  "message": "Replace archetype structure with new simplified one."
```
Comment (vladiIkor) [inline changelogs/feature/new_adapter_structure.json]: This is very important change, we should be as precise as possible, hints:

- It's a new adapter structure, not archetype. 
- Switching from multi module adapters to no modules application structure

File: docs/README.md [hunk 3]
```diff
@@ -39,16 +39,15 @@ systems, but all the principles apply equally to such scenarios.
...
 
 ![Image of SIP Adapter](./img/SIP_readme_adapter_detail.svg?raw=true "SIP Adapter")
 
 **Domain**
```
Comment (vladiIkor) [inline docs/README.md]: Domain or Common?

File: docs/README.md [hunk 4]
```diff
@@ -39,16 +39,15 @@ systems, but all the principles apply equally to such scenarios.
...
 
-The **Domain** module takes a central position in the image, because it provides the common data model for both systems.
-Also, it separates the system connectors from each other to allow loose coupling between them. It should not contain any
+The **Common** Package takes a central position in the image, 
+because it provides the common data model inside domain for both systems and common util functionalities.
+Also, it separates the system connectors from each other to allow loose coupling between them. Domain package should not contain any
```
Comment (vladiIkor) [inline docs/README.md]: 'Also, it separates the system connectors from each other to allow loose coupling between them.' - not true anymore, it does not separate anything and does not give loose coupling.

The remaining text is true, but I'm afraid it becomes hard to understand in current format. The sub title is Common, and we start talking about common package than all of a sudden the story switches to domain package.

File: docs/README.md [hunk 5]
```diff
@@ -179,110 +174,31 @@ can keep their independence.
...
+│   │   ├───domain
+│   │   └───util
+│   ├───connectors
+│   │   └───connector1
+│   │       ├───config
+│   │       ├───transformer
+│   │       ├───processors
+│   │       ├───sink
+│   │       ├───validators
+│   │       └───domain
```
Comment (vladiIkor) [inline docs/README.md]: connector2?

File: docs/README.md [hunk 6]
```diff
@@ -39,16 +39,15 @@ systems, but all the principles apply equally to such scenarios.
 The image below shows the modules structure of a single SIP adapter.
```
Comment (vladiIkor) [inline docs/README.md]: The image needs to be updated as well. Currently it shows dependencies between modules etc.

**Review by vladiIkor — DISMISSED** [review]

File: sip-archetype/src/main/resources/archetype-resources/pom.xml [hunk 7]
```diff
@@ -31,23 +24,45 @@
             <artifactId>lombok</artifactId>
         </dependency>
 #end
+        <dependency>
+            <groupId>de.ikor.sip.foundation</groupId>
+            <artifactId>sip-integration-starter</artifactId>
+        </dependency>
+        <!-- test dependencies -->
+        <dependency>
```
Comment (vladiIkor) [inline sip-archetype/src/main/resources/archetype-resources/pom.xml]: If we want this dependency on the each adapter (which seams like a reasonable choice), we should add the dependency to sip-integration-starter.

**Review by LetoBukarica — APPROVED** [review]

**Pull request merged**